File: Difficulty: Easy/Minimum number of Coins/minimum-number-of-coins.cpp
class Solution {
  public:
    int findMin(int N) {
        // code here
        vector<int>coins{10,5,2,1};
        vector<int>an;
        int i=0;
        while(N)
        {
            int count=N/coins[i];
            while(count--){
               an.push_back(coins[i]); 
            }
            N%=coins[i];
            i++;
        }
        return an.size();
    }
};

File: Difficulty: Easy/Minimum number of Coins/minimum-number-of-coins.js
/**
 * @param {number} n
 * @returns {number}
 */
class Solution {
    findMin(n) {
        let currency=[10,5,2,1];
        let ans=[];
        let i=0;
        while(n>0)
        {
            let count=Math.floor(n/currency[i]);
            while(count>0)
            {
                ans.push(currency[i]);
                count--;
            }
            n=n%currency[i];
            i++;
        }
        return ans.length;
    }
}